src/main/java/org/fogbowcloud/arrebol/execution/k8s/K8sTaskExecutor.java

```diff
@@ -0,0 +1,166 @@
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Integer id;
+	@Transient
+	private final Logger LOGGER = Logger.getLogger(K8sTaskExecutor.class);
+	@OneToOne(cascade = CascadeType.ALL, orphanRemoval = true, targetEntity = DefaultK8sClusterResource.class)
+	private K8sClusterResource k8sClusterResource;
+	@OneToOne(cascade = CascadeType.ALL, orphanRemoval = true, targetEntity = DefaultK8sClient.class)
+	private K8sClient k8sClient;
+	@Transient
+	private static final long POOLING_PERIOD_TIME_MS = 5 * 1000;
+
+	public K8sTaskExecutor(K8sClusterResource k8sClusterResource, K8sClient k8sClient) {
+		this.k8sClusterResource = k8sClusterResource;
+		this.k8sClient = k8sClient;
+	}
+
+	public K8sTaskExecutor() {
+	}
+
+	@Override
+	public TaskExecutionResult execute(Task task) {
+		TaskExecutionResult taskExecutionResult = null;
+
+		String jobName = task.getId();
+
+		List<Command> commands = task.getTaskSpec().getCommands();
+		String command = joinCommands(commands);
+
+		Map<String, String> requirements = task.getTaskSpec().getRequirements();
+		String imageId = getImageId(requirements);
+
+		LOGGER.debug("Image id: " + imageId);
+		LOGGER.debug("Command: " + command);
+
+		int tasksListSize = task.getTaskSpec().getCommands().size();
+
+		try {
+			V1Job job = k8sClient.createJob(jobName, imageId, command);
+			boolean jobIsRunning = true;
+
+			while (jobIsRunning) {
+				try {
+					sleep(POOLING_PERIOD_TIME_MS);
+				} catch (InterruptedException e) {
+					LOGGER.error(e.getMessage(), e);
+				}
+
+				job = k8sClient.getJob(jobName);
+				jobIsRunning = job.getStatus().getActive() != null;
+			}
+
+			if (wasSuccessful(job)) {
+				successTask(task);
+				taskExecutionResult = getSuccessResultInstance(tasksListSize);
+			} else {
+				failTask(task);
+				taskExecutionResult = getFailResultInstance(tasksListSize);
+			}
+		} catch (ApiException e) {
+			LOGGER.error("Error while call K8s client API. " + e.getMessage(), e);
+			failTask(task);
+			taskExecutionResult = getFailResultInstance(tasksListSize);
+			return taskExecutionResult;
+		}
+
+		return taskExecutionResult;
+	}
+
+	@Override
+	public Map<String, String> getMetadata() {
+		Map<String, String> metadata = new HashMap<>();
+		String address = this.k8sClusterResource.getApiAddress();
+		metadata.put(ADDRESS_METADATA_KEY, address);
+		return metadata;
+	}
+
+	private String joinCommands(List<Command> commands) {
+		String result = "";
+		for (Command command : commands)
+			result += command.getCommand() + " && ";
+		result = result.substring(0, result.length() - 4);
+
+		return result;
+	}
+
+	private String getImageId(Map<String, String> requirements) throws IllegalArgumentException {
+		String imageId = null;
+		if (Objects.nonNull(requirements))
+			imageId = requirements.get(RequirementsContants.IMAGE_KEY);
+
+		return imageId;
+	}
+
+	private boolean wasSuccessful(V1Job job) {
+		Integer successedQuant = job.getStatus().getSucceeded();
```

Comment:
I think you mean succeededAmount, right?

Reply:
Yes, I updated